go/workspace.rst

```diff
@@ -0,0 +1,208 @@
+.. _github.com/golang/protobuf: https://github.com/golang/protobuf/
+.. _google.golang.org/grpc: https://github.com/grpc/grpc-go
+.. _golang.org/x/net: https://github.com/golang/net/
+.. _golang.org/x/tools: https://github.com/golang/tools/
+.. _go_library: core.rst#go_library
+.. _toolchains: toolchains.rst
+.. _go_register_toolchains: toolchains.rst#go_register_toolchains
+.. _go_sdk: toolchains.rst#go_sdk
+.. _go_toolchain: toolchains.rst#go_toolchain
+.. _new_go_repository: deprecated.rst#new_go_repository
+.. _normal go logic: https://golang.org/cmd/go/#hdr-Remote_import_paths
+.. _gazelle: tools/gazelle/README.md
+.. _http_archive: https://docs.bazel.build/versions/master/be/workspace.html#http_archive
+.. _git_repository: https://docs.bazel.build/versions/master/be/workspace.html#git_repository
+.. _nested workspaces: https://bazel.build/designs/2016/09/19/recursive-ws-parsing.html
+
+.. _go_prefix_faq: /README.rst#whats-up-with-the-go_default_library-name
+.. |go_prefix_faq| replace:: FAQ
+
+.. role:: param(kbd)
+.. role:: type(emphasis)
+.. role:: value(code)
+.. |mandatory| replace:: **mandatory value**
+
+Workspace rules are either repository rules, or macros that are intended to be used from the
+WORKSPACE file.
+
+`Toolchain support <toolchains>`_:
+    This is both the functions to add the normal go toolchains (go_register_toolchains_) and the
+    ones to declare your own toolchains (go_sdk_ and go_toolchain_).
+    These are here for the long term, and are documented in the toolchains_ documentation.
+
+go_rules_dependencies_:
+    This is invoked to add the things that the go rules depend on to your workspace.
+    We expect this to be replaced by a Bazel feature (`nested workspaces`_) in the long term.
+
+go_repository_:
+    This is used to automatically generate build files for a go repository that you depend on.
+    In the future we expect this to be replaced by normal http_archive_ or git_repository_ rules,
+    once gazelle_ fully supports flat build files.
+    There is also the deprecated new_go_repository_ which you should no longer use (we will be
+    deleting it soon).
+
+go_prefix_:
+    This is a legacy from when the import path for a go_library_ was determined from the root
+    go_prefix and the path from the workspace root. Now instead we have every single go_library_
+    know it's own import path. We currently maintain this rule for backwards compatability, but we
+    expect to have it removed well before 1.0
+
+-----
+
+go_rules_dependencies
+~~~~~~~~~~~~~~~~~~~~~
+
+Adds Go-related external dependencies to the WORKSPACE, including the Go
+toolchain and standard library. All the other workspace rules and build rules
+assume that this rule is placed in the WORKSPACE.
+
+When `nested workspaces`_  arrive this will be redundant, but for nowyou should **always** call this macro from your WORKSPACE.
+
+It only adds repositories that have not previously been declared, so you can override anything it
```

Comment:
I think this paragraph should describe how to override a dependency. Consider moving it below the list of dependencies, rewording, and including an example. Something like:

"If you want to use a different version of one of these repository, just declare it in your WORKSPACE file *before* calling `go_rules_dependencies()`. This macro will only declare  a repository if it hasn't already been declared."